File: convert_to_base64.py
import base64
import sys

# 检查命令行参数
if len(sys.argv) != 2:
    print("用法: python convert_to_base64.py <图片文件路径>")
    sys.exit(1)

# 获取图片文件路径
image_path = sys.argv[1]

# 获取文件扩展名
extension = image_path.split('.')[-1].lower()
if extension == 'jpg':
    extension = 'jpeg'  # MIME类型使用jpeg而不是jpg

# 转换为base64
with open(image_path, 'rb') as image_file:
    base64_bytes = base64.b64encode(image_file.read())
    base64_string = base64_bytes.decode('utf-8')

# 生成data URI
data_uri = f"data:image/{extension};base64,{base64_string}"

# 输出结果
print("=== Base64 Data URI ===")
print(data_uri)
print("\n=== HTML Img Tag ===")
print(f'<img src="{data_uri}" alt="Image">')
print("\n=== CSS Background ===")
print(f'background-image: url("{data_uri}");')

# 保存到文件
output_file = image_path.replace(f'.{extension}', '_base64.txt')
with open(output_file, 'w', encoding='utf-8') as f:
    f.write("=== Base64 Data URI ===\n")
    f.write(data_uri)
    f.write("\n\n=== HTML Img Tag ===\n")
    f.write(f'<img src="{data_uri}" alt="Image">\n')
    f.write("\n=== CSS Background ===\n")
    f.write(f'background-image: url("{data_uri}");\n')

print(f"\n结果已保存到: {output_file}")

File: convert_to_base64.js
// Node.js图片转Base64转换器
// 使用方法: node convert_to_base64.js "E:\lyl-Ollamawebui\logo\logo.jpg"

const fs = require('fs');
const path = require('path');

const imagePath = process.argv[2];

if (!imagePath) {
    console.error('请提供图片路径');
    console.error('使用方法: node convert_to_base64.js "图片路径"');
    process.exit(1);
}

// 检查文件是否存在
if (!fs.existsSync(imagePath)) {
    console.error(`文件不存在: ${imagePath}`);
    process.exit(1);
}

// 获取文件扩展名
const extension = path.extname(imagePath).toLowerCase();

// 映射文件扩展名到MIME类型
const mimeTypes = {
    '.jpg': 'image/jpeg',
    '.jpeg': 'image/jpeg',
    '.png': 'image/png',
    '.gif': 'image/gif',
    '.bmp': 'image/bmp',
    '.webp': 'image/webp'
};

if (!mimeTypes[extension]) {
    console.error(`不支持的图片格式: ${extension}`);
    process.exit(1);
}

const mimeType = mimeTypes[extension];

// 读取文件并转换为Base64
const buffer = fs.readFileSync(imagePath);
const base64String = buffer.toString('base64');

// 构建完整的Base64数据URI
const base64Uri = `data:${mimeType};base64,${base64String}`;

// 输出结果
console.log(`图片路径: ${imagePath}`);
console.log(`MIME类型: ${mimeType}`);
console.log('Base64数据URI:');
console.log(base64Uri);

// 保存到文件
const outputFile = path.join(path.dirname(imagePath), `${path.basename(imagePath, extension)}_base64.txt`);
fs.writeFileSync(outputFile, base64Uri, 'utf8');
console.log(`结果已保存到: ${outputFile}`);
